src/OrbitGl/TimeGraph.cpp

```diff
@@ -127,6 +127,9 @@ double TimeGraph::GetCurrentTimeSpanUs() const { return max_time_us_ - min_time_
 void TimeGraph::ZoomTime(float zoom_value, double mouse_ratio) {
   static double increment_ratio = 0.1;
   double scale = (zoom_value > 0) ? (1 + increment_ratio) : (1 / (1 + increment_ratio));
+  // The horizontal zoom could have been triggered from the margin of TimeGraph, so we clamp the
+  // mouse_ratio to ensure it is between 0 and 1.
+  mouse_ratio = std::clamp(mouse_ratio, 0., 1.);
```

Comment:
interesting... why could it fall outside of (0,1]? Also is 0 valid value?

Reply:
Now, for the TimeGraph's time range we are having only the visible place (not including the right slider) for making the max_ and min_.

So, the question is what to do if an user wants to zoom having the mouse in the right_slider. I think we have 2 options:
- Not allow zooming in the right_slider
- Making this kind of zooms as they were from the rightmost point in the timerange.

Without this clamp, the zoom is a bit strange and not reversible I believe.

I chose the second one because I think that zoom could be useful from the right_margin.

And 0 is a valid value, the invariant here is that the ratio (max-current)/(current-min) is always the same but we are not making any division there, so it works for [0,1].